src/main/java/com/ericsson/ei/subscriptionhandler/InformSubscriber.java

```diff
@@ -0,0 +1,427 @@
+import com.mongodb.BasicDBObject;
+
+import lombok.Getter;
+
+/**
+ * This class represents the REST POST notification mechanism and the alternate
+ * way to save the aggregatedObject details in the database when the
+ * notification fails.
+ *
+ * @author xjibbal
+ */
+
+@Component
+public class InformSubscriber {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InformSubscriber.class);
+    // Regular expression for replacement unexpected character like \"|
+    private static final String REGEX = "^\"|\"$";
+    private static final String JENKINS_CRUMB_ENDPOINT = "/crumbIssuer/api/json";
+
+    @Getter
+    @Value("${notification.failAttempt:#{0}}")
+    private int failAttempt;
+
+    @Getter
+    @Value("${missedNotificationCollectionName}")
+    private String missedNotificationCollectionName;
+
+    @Getter
+    @Value("${missedNotificationDataBaseName}")
+    private String missedNotificationDataBaseName;
+
+    @Getter
+    @Value("${notification.ttl.value}")
+    private int ttlValue;
+
+    @Autowired
+    private JmesPathInterface jmespath;
+
+    @Autowired
+    private SpringRestTemplate restTemplate;
+
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+
+    @Autowired
+    private SendMail sendMail;
+
+    /**
+     * This method extracts the mode of notification through which the
+     * subscriber should be notified, from the subscription Object. And if the
+     * notification fails, then it saved in the database.
+     *
+     * @param aggregatedObject
+     * @param subscriptionJson
+     */
+    public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson) {
+        String subscriptionName = getSubscriptionField("subscriptionName", subscriptionJson);
+        String notificationType = getSubscriptionField("notificationType", subscriptionJson);
+        String notificationMeta = getSubscriptionField("notificationMeta", subscriptionJson);
+
+        String subject = getSubscriptionField("emailSubject", subscriptionJson);
+
+        MultiValueMap<String, String> mapNotificationMessage = mapNotificationMessage(aggregatedObject,
+                subscriptionJson);
+
+        if (notificationType.trim().equals("REST_POST")) {
+            LOGGER.debug("Notification through REST_POST");
+            boolean success = false;
+
+            if (notificationMeta.contains("?")) {
+                LOGGER.debug("Unformatted notificationMeta = " + notificationMeta);
+                try {
+                    notificationMeta = replaceParamsValuesWithAggregatedData(aggregatedObject, notificationMeta);
+                    LOGGER.debug("Formatted notificationMeta = " + notificationMeta);
+                } catch (URISyntaxException e) {
+                    LOGGER.error("Failed to reformat url reason: " + e.getMessage());
+                }
+            }
+
+            // Prepare request headers
+            HttpHeaders headers = new HttpHeaders();
```

Comment:
move header constructor inside prepeareHeaders, remove it as argument.

Reply:
Done.